### PR TITLE
frontend: move create AI Agent form to use Field component

### PR DESCRIPTION
**Pull request by malinskibeniamin**
Form component migration to new Field pattern:
  - Migrated AI Agent create form (3 files)
  - Migrated SecretSelector component
  - Migrated UserSelector component

  Added form-refactorer skill with migration docs & patterns

  Total: 8 files changed, replaced FormField/FormItem/FormLabel/FormMessage with Field/FieldLabel/FieldDescription/FieldError across forms

### CODE REVIEW

File: frontend/.claude/skills/form-refactorer/SKILL.md
Comment (malinskibeniamin) [inline frontend/.claude/skills/form-refactorer/SKILL.md:1]: this skill will be part of the UI registry in the future.

File: frontend/src/components/pages/agents/create/ai-agent-create-page.tsx [hunk 1]
```diff
@@ -487,19 +478,21 @@ export const AIAgentCreatePage = () => {
...
+                <Field data-invalid={!!form.formState.errors.systemPrompt}>
+                  <FieldLabel htmlFor="systemPrompt" required>
+                    System Prompt
+                  </FieldLabel>
+                  <Textarea
+                    id="systemPrompt"
+                    placeholder="You are a helpful AI agent that..."
+                    rows={8}
+                    {...form.register('systemPrompt')}
+                    aria-invalid={!!form.formState.errors.systemPrompt}
```
Comment (jvorcak) [inline frontend/src/components/pages/agents/create/ai-agent-create-page.tsx:493]: nit: should we add `aria-describedby` if we add `aria-invalid` if we can, since we're doing it properly now?

File: frontend/src/components/pages/agents/create/ai-agent-create-page.tsx [hunk 2]
```diff
@@ -510,34 +503,31 @@ export const AIAgentCreatePage = () => {
...
+                    render={({ field }) =>
+                      availableMcpServers.length > 0 ? (
+                        <MCPServerCardList
+                          idPrefix="root"
+                          onValueChange={field.onChange}
+                          servers={availableMcpServers}
+                          value={field.value || []}
+                        />
+                      ) : (
+                        <MCPEmpty>
```
Comment (jvorcak) [inline frontend/src/components/pages/agents/create/ai-agent-create-page.tsx]: Shall we add test-ids for these empty states for e2e?

File: frontend/src/components/ui/ai-agent/llm-config-section.tsx [hunk 3]
```diff
@@ -121,48 +116,49 @@ export const LLMConfigSection: React.FC<LLMConfigSectionProps> = ({
...
-            );
-          }
+            if (isFreeTextMode) {
+              return (
+                <>
+                  <Input
+                    id="model"
+                    placeholder="Enter model name (e.g., llama-3.1-70b)"
+                    {...field}
+                    aria-invalid={!!form.formState.errors[fieldNames.model]}
```
Comment (jvorcak) [inline frontend/src/components/ui/ai-agent/llm-config-section.tsx:141]: The same here, `aria-describedby` is doable I think.
